src/garbage-collection.md

```diff
@@ -0,0 +1,105 @@
+---
+description: Everything about garbage collection in AssemblyScript.
+sidebarDepth: 0
+---
+
+# Garbage Collection
+
+::: tip NOTE
+This section covers the new runtime added in AssemblyScript 0.18. The previous runtime documentation [can be found here](./runtime.md).
+:::
+
+While the WebAssembly [GC](https://github.com/WebAssembly/gc) 🦄 proposal is still in the works, AssemblyScript ships with multiple garbage collector runtimes implemented on top of linear memory that are useful for different use cases. Their interface is always the same, so these are interchangeable, yet differ in their level of sophistication.
+
+## Runtime variants
+
+With the `--runtime` compiler option, one of multiple runtime variantes can be selected, which determines the memory allocator and garbage collector included in a program:
+
+```
+  --runtime             Specifies the runtime variant to include in the program.
+
+                         default      TLSF + lightweight GC invoked externally
+                         incremental  TLSF + incremental GC (experimental)
+                         stub         Minimal runtime stub (never frees)
+                         ...          Path to a custom runtime implementation
+```
+
+Using the `--exportRuntime` compiler option, the runtime interface can be exported to the host to enable external allocation etc.
+
+```
+  --exportRuntime       Exports the runtime helpers (__new, __collect etc.).
+```
+
+To interface with the runtime externally, you'll probably want to use the [loader](./loader.md) as well.
+
+## "Default" runtime
+
+The default runtime provides a complete dynamic memory manager backing `new`, `heap.alloc`, `heap.realloc` and `heap.free` and is suitable for most non-interactive use cases, like the compiler itself.
```

Comment:
I'd maybe give an example on what you mean by `non-interactive` (E.g, stateful services that use input to perform actions? Like Games, or video decoders, etc... ?)

Reply:
Added more prose to that paragraph, explaining why it is a good fit for the compiler or similar use cases.